src/Onboarding/Routes/LocationRoute.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Give\Onboarding\Routes;
+
+use WP_REST_Request;
+use Give\API\RestRoute;
+use Give\Onboarding\Helpers\FormatList;
+
+class LocationRoute implements RestRoute {
+
+	protected $endpoint = 'onboarding/location';
+
+	public function handleRequest( WP_REST_Request $request ) {
+
+		$countryCode = $request->get_param( 'countryCode' );
+		$statesList  = give_get_states( $countryCode );
+
+		$statesList[''] = '-';
+
+		return [
+			'states' => FormatList::fromKeyValue( $statesList ),
+		];
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function registerRoute() {
+		register_rest_route(
+			'give-api/v2',
+			$this->endpoint,
+			[
+				[
+					'methods'             => 'GET',
+					'callback'            => [ $this, 'handleRequest' ],
+					'permission_callback' => function() {
+						return current_user_can( 'manage_options' );
+					},
+					'args'                => [
+						'countryCode' => [
+							'type'              => 'string',
+							'required'          => true,
+							// 'validate_callback' => [ $this, 'validateSetting' ],
```

Comment:
I think you should implement a validation callback here, which just compares the request parameter against an array of accepted country codes.

Reply:
Ah, yeah. Good catch!

Reply:
Resolved by 8213acaa6